crates/wasm-mutate/src/mutators/peephole/mod.rs

```diff
@@ -212,7 +212,7 @@ impl PeepholeMutator {
                     config.info().function_map.clone(),
                 );
                 let runner = Runner::<Lang, PeepholeMutationAnalysis, ()>::new(analysis)
-                    .with_iter_limit(1) // only one iterations, do not wait for eq saturation, increasing only by one it affects the execution time of the mutator by a lot
+                    .with_iter_limit(200) // FIXME, the iterations should consume fuel from the actual mutator. Be careful with inner set time limits that can lead us to non-deterministic behavior
```

Comment:
Here is the source of the long running test. Actually we should link the iteration of the equality saturation to the fuel.

Reply:
Still, I think this will be solved for `wasm-shrink` by adding only shrinking-based rules if `reduce` is true for the configuration

Reply:
And also we should switch from using the random extractor to using the `AstSize` extractor when `reduce(true)`